src/history/hs.c

```diff
@@ -1029,19 +1027,17 @@ __wt_find_hs_upd(WT_SESSION_IMPL *session, WT_ITEM *key, uint64_t recno, WT_UPDA
         WT_STAT_CONN_INCR(session, cache_hs_read_squash);
     }
 
-    /* Allocate an update structure for the record found. */
-    WT_ERR(__wt_upd_alloc(session, hs_value, upd_type, &upd, NULL));
-    upd->txnid = WT_TXN_NONE;
-    upd->durable_ts = durable_timestamp;
-    upd->start_ts = hs_start_ts;
-    upd->prepare_state = upd->start_ts == upd->durable_ts ? WT_PREPARE_INIT : WT_PREPARE_RESOLVED;
-
     /*
-     * We're not keeping this in our update list as we want to get rid of it after the read has been
-     * dealt with. Mark this update as external and to be discarded when not needed.
+     * Potential optimization: We can likely get rid of this copy and the update allocation above.
```

Comment:
If we did the other optimization I've alluded to and we had something like `WT_ITEM_VECTOR`, we could remove this.